src/main/scala/uk/gov/nationalarchives/api/update/Lambda.scala

```diff
@@ -47,30 +46,79 @@ class Lambda {
-            processor.process(ffidMetadataInput, bodyWithReceiptHandle.receiptHandle)
-        }.left.map(circeError =>
-          FailedApiUpdateException(bodyWithReceiptHandle.receiptHandle, circeError)
-        )
-      }).toList
-
-    Await.result(ResultCollector(config, sqsUtils).collect(results), 10 seconds)
+  def getInputs(inputStream: InputStream): Future[Input] = {
+    val body = Source.fromInputStream(inputStream).getLines().mkString
+    decode[Input](body) match {
+      case Left(error) => Future.failed(error)
+      case Right(input) => Future.successful(input)
+    }
+  }
+
+  def getFileMetadata(input: Input): List[AddFileMetadataWithFileIdInputValues] = {
+    input.results.flatMap(_.fileCheckResults.checksum.map(c => AddFileMetadataWithFileIdInputValues("SHA256ServerSideChecksum", c.fileId, c.sha256Checksum))) ++
+    input.redactedResults.redactedFiles.map(r => AddFileMetadataWithFileIdInputValues("OriginalFilepath", r.redactedFileId, r.originalFilePath))
+  }
+
+  def sendStatuses(input: Input, token: BearerAccessToken): Future[Statuses] = {
+    val (consignmentStatuses, fileStatuses) = input.statuses.statuses.partition(_.statusType == "Consignment")
+
+
+    for {
+      _ <- Future.sequence {
+        consignmentStatuses.map(consignmentStatus => {
+          val consignmentStatusVariables = acs.Variables(ConsignmentStatusInput(consignmentStatus.id, consignmentStatus.statusName, Option(consignmentStatus.statusValue)))
+          RequestSender[acs.Data, acs.Variables].sendRequest(token, acs.document, consignmentStatusVariables)
+        })
+      }
+      _ <- if (fileStatuses.nonEmpty) {
+        val fileStatusVariables = amfs.Variables(AddMultipleFileStatusesInput(fileStatuses.map(fs => AddFileStatusInput(fs.id, fs.statusName, fs.statusValue))))
+        RequestSender[amfs.Data, amfs.Variables].sendRequest(token, amfs.document, fileStatusVariables)
+      } else {
+        Future()
+      }
+    } yield input.statuses
   }
+
+  def update(inputStream: InputStream, output: OutputStream): Unit = {
+    val result = for {
+      input <- getInputs(inputStream)
+      token <- keycloakUtils.serviceAccountToken(config("client.id"), clientSecret)
+      av: avbm.Data <- RequestSender[avbm.Data, avbm.Variables]().sendRequest(token, avbm.document, avbm.Variables(AddAntivirusMetadataInput(input.results.flatMap(_.fileCheckResults.antivirus))))
+      file: amfm.Data <- RequestSender[amfm.Data, amfm.Variables].sendRequest(token, amfm.document, amfm.Variables(AddFileMetadataWithFileIdInput(getFileMetadata(input))))
+      ffid: abfim.Data <- RequestSender[abfim.Data, abfim.Variables].sendRequest(token, abfim.document, abfim.Variables(FFIDMetadataInput(input.results.flatMap(_.fileCheckResults.fileFormat))))
+      statuses <- sendStatuses(input, token)
+    } yield {
+      output.write(Result(av, file, ffid, statuses).asJson.printWith(Printer.noSpaces).getBytes())
+    }
+    Await.result(result, 60.seconds)
```

Comment:
Is there a reason for the `Await`?

Reply:
Yeah, AWS lambda doesn't understand Scala futures, if you don't put the await in, it will terminate before it's had a chance to do anything.